source/base/ReferenceCounter.ooc

```diff
@@ -1,20 +1,29 @@
 import threading/Thread
 import Synchronized
 
-ReferenceCounter: class extends Synchronized {
+ReferenceCounter: class {
 	_target: Object
 	count: static Int = 0
 	_count: Int = 0
 	_kill := false
+	_lock := Mutex new(MutexType Safe)
```

Comment:
How about we modify the constructor so it takes a bool that defaults to true instead? That way the Mutex won't have to be created just to be destroyed immediately afterwards...